molgenis-metadata-manager/src/main/frontend/src/store/actions.js

```diff
@@ -0,0 +1,192 @@
+        })
+      })
+  },
+  /**
+   * Retrieve all EntityTypes and filter on non-system EntityTypes
+   */
+  [GET_ENTITY_TYPES] ({commit}) {
+    // TODO can we filter system entities with REST call??
+    get({apiUrl: '/api'}, '/v2/sys_md_EntityType?num=10000')
+      .then(response => {
+        commit(SET_ENTITY_TYPES, response.items)
+      }, error => {
+        commit(CREATE_ALERT, {
+          type: 'error',
+          message: error.errors[0].message
+        })
+      })
+  },
+  /**
+   * Retrieve all Attribute types
+   */
+  [GET_ATTRIBUTE_TYPES] ({commit}) {
+    get({apiUrl: '/api'}, '/v2/sys_md_Attribute/meta/type')
+      .then(response => {
+        commit(SET_ATTRIBUTE_TYPES, response.enumOptions.map((type) => type.toUpperCase()))
+      }, error => {
+        commit(CREATE_ALERT, {
+          type: 'error',
+          message: error.errors[0].message
+        })
+      })
+  },
+  /**
+   * Retrieve EditorEntityType based on EntityType ID
+   *
+   * @param entityTypeId The selected EntityType identifier
+   */
+  [GET_EDITOR_ENTITY_TYPE] ({commit}, entityTypeId) {
+    get({apiUrl: '/metadata-manager-service'}, '/entityType/' + entityTypeId)
+      .then(response => {
+        commit(SET_EDITOR_ENTITY_TYPE, toEntityType(response.entityType))
+      }, error => {
+        commit(CREATE_ALERT, {
+          type: 'error',
+          message: error.errors[0].message
+        })
+      })
+  },
+  /**
+   * Create a new EntityType
+   * Response returns a blank EditorEntityType
+   * EditorEntityType is added to the list of entityTypes in the state
+   */
+  [CREATE_ENTITY_TYPE] ({commit}) {
+    get({apiUrl: '/metadata-manager-service'}, '/create/entityType')
+      .then(response => {
+        const newEditorEntityType = toEntityType(response.entityType)
+        newEditorEntityType.isNew = true
+        commit(SET_EDITOR_ENTITY_TYPE, newEditorEntityType)
+      }, error => {
+        if (error.errors) {
+          commit(CREATE_ALERT, {
+            type: 'error',
+            message: error.errors[0].message
+          })
+        } else {
+          commit(CREATE_ALERT, {
+            type: 'error',
+            message: 'Something went wrong, make sure you have permissions for creating entities.'
+          })
+        }
+      })
+  },
+  /**
+   * Deletes an EntityType and reloads the EntityTypes present in the state
+   *
+   * @param selectedEntityTypeId the ID of the EntityType to be deleted
+   */
+  [DELETE_ENTITY_TYPE] ({commit, state}, selectedEntityTypeId) {
+    callApi({apiUrl: '/api'}, '/v1/' + selectedEntityTypeId + '/meta', 'delete')
+      .then(response => {
+        // Never reached due to https://github.com/molgenis/molgenis-api-client/issues/1
+      }, error => {
+        if (error.errors) {
+          commit(CREATE_ALERT, {
+            type: 'error',
+            message: error.errors[0].message
+          })
+        } else {
+          // Clear selected editorEntityType
```

Comment:
i don't think we need all of these code comments here , the const names seem clear to me